menuinst/platforms/win_utils/registry.py

```diff
@@ -107,16 +107,10 @@ def unregister_file_extension(extension, identifier, mode="user"):
         with winreg.OpenKey(
             root, rf"Software\Classes\{extension}\OpenWithProgids", 0, winreg.KEY_ALL_ACCESS
         ) as key:
-            try:
-                winreg.QueryValueEx(key, identifier)
-            except FileNotFoundError:
-                log.debug(
-                    "Handler '%s' is not associated with extension '%s'", identifier, extension
-                )
-            else:
-                winreg.DeleteValue(key, identifier)
-    except Exception as exc:
-        log.exception("Could not check key '%s' for deletion", extension, exc_info=exc)
+            winreg.QueryValueEx(key, identifier)  # ??? Is this necessary?
```

Comment:
This is probably not necessary since `winreg.DeleteValue` already raises an `OSError`.

Could you add a test for this inside `test_windows_registry.py`? The `capsys` fixture should be good to capture the output.

Reply:
You mean to test for `OSError`? Not sure what "test for this" refers to.

Reply:
That's right. It looks like [tests/test_windows_registry.py](https://github.com/conda/menuinst/blob/d0fce331068023722aece0c5d6a1479b20c5ea29/tests/test_windows_registry.py) has tests for successful runs, but not for failures.